### PR TITLE
feat(dashboard): implementing-e2e-tests-on-dashboard-permissions-check

### PR DESCRIPTION
**Pull request by paul-oureib**
## Description

Implementing the E2E tests from "Dashboards - ACL Management - Check permissions" test case.*

(Sidenote: The shareDashboardToUser command is implemented and is confirmed to work on Cypress locally, but for some reason it just couldn't do what I wanted on the CI, making the tenth scenario fail inevitably, hence the implementation in the Given of the tenth scenario to do via the UI what it is supposed to take care of. I could not figure out the reason for it not working, but I left the command as-is nonetheless because it could still be useful in the future and could very well be an efficient tool for future refactorings)

**Fixes** # MON-20507

**Ref** # MON-19999

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by bdauria — APPROVED** [review]

File: centreon/packages/js-config/cypress/e2e/commands.ts [hunk 1]
```diff
@@ -391,6 +391,42 @@ Cypress.Commands.add(
   }
 );
 
+interface ShareDashboardToUserProps {
+  dashboardName: string;
+  role: string;
+  userName: string;
+}
+
+Cypress.Commands.add(
```
Comment (Thebarda) [inline centreon/packages/js-config/cypress/e2e/commands.ts:400]: You can move this into the commands file in centreon

File: centreon/tests/e2e/cypress/e2e/Dashboards/05-dashboard-check-permissions/index.ts [hunk 2]
```diff
@@ -0,0 +1,783 @@
...
+  }).as('listAllDashboards');
+  cy.intercept({
+    method: 'POST',
+    url: '/centreon/api/latest/configuration/dashboards'
+  }).as('createDashboard');
+  cy.loginByTypeOfUser({
+    jsonName: adminUser.login,
+    loginViaApi: true
+  });
+  cy.insertDashboard({ ...dashboards.fromAdminUser });
```
Comment (Thebarda) [inline centreon/tests/e2e/cypress/e2e/Dashboards/05-dashboard-check-permissions/index.ts:34]:
```suggestion
  cy.insertDashboard(dashboards.fromAdminUser);
```
This should work too

File: centreon/tests/e2e/cypress/e2e/Dashboards/05-dashboard-check-permissions/index.ts [hunk 3]
```diff
@@ -0,0 +1,783 @@
...
+  });
+  cy.insertDashboard({ ...dashboards.fromDashboardAdministratorUser });
+  cy.logoutViaAPI();
+  cy.loginByTypeOfUser({
+    jsonName: dashboardCreatorUser.login,
+    loginViaApi: true
+  });
+  cy.insertDashboard({ ...dashboards.fromDashboardCreatorUser });
+  cy.logoutViaAPI();
+});
```
Comment (Thebarda) [inline centreon/tests/e2e/cypress/e2e/Dashboards/05-dashboard-check-permissions/index.ts:48]:
```suggestion
  cy.insertDashboard(dashboards.fromDashboardAdministratorUser);
  cy.logoutViaAPI();
  cy.loginByTypeOfUser({
    jsonName: dashboardCreatorUser.login,
    loginViaApi: true
  });
  cy.insertDashboard(dashboards.fromDashboardCreatorUser);
  cy.logoutViaAPI();
});
```

File: centreon/tests/e2e/cypress/e2e/Dashboards/05-dashboard-check-permissions/index.ts [hunk 4]
```diff
@@ -0,0 +1,783 @@
...
+
+Then(
+  'the dashboard administrator user is redirected to the detail page for this dashboard',
+  () => {
+    cy.location('pathname')
+      .should('include', '/dashboards/')
+      .invoke('split', '/')
+      .should('not.be.empty')
+      .then(Cypress._.last)
```
Comment (Thebarda) [inline centreon/tests/e2e/cypress/e2e/Dashboards/05-dashboard-check-permissions/index.ts:319]: I wonder if the `last` function from ramda works here

**Review by bdauria — APPROVED** [review]

**Pull request merged**